clippy_dev/src/new_lint.rs

```diff
@@ -0,0 +1,184 @@
+use std::fs::{File, OpenOptions};
+use std::io;
+use std::io::prelude::*;
+use std::io::ErrorKind;
+use std::path::{Path, PathBuf};
+
+pub fn create(pass: Option<&str>, lint_name: Option<&str>, category: Option<&str>) -> Result<(), io::Error> {
+    let pass = pass.expect("`pass` argument is validated by clap");
+    let lint_name = lint_name.expect("`name` argument is validated by clap");
+    let category = category.expect("`category` argument is validated by clap");
+
+    match open_files(lint_name) {
+        Ok((mut test_file, mut lint_file)) => {
+            let (pass_type, pass_import, context_import) = match pass {
+                "early" => ("EarlyLintPass", "use syntax::ast::*;", "EarlyContext"),
+                "late" => ("LateLintPass", "use rustc_hir::*;", "LateContext"),
+                _ => {
+                    unreachable!("`pass_type` should only ever be `early` or `late`!");
+                },
+            };
+
+            let camel_case_name = to_camel_case(lint_name);
+
+            match test_file.write_all(get_test_file_contents(lint_name).as_bytes()) {
```

Comment:
You can changes this to an `if let Err(e) = test_file.write_all(get_test_file_contents(lint_name).as_bytes()) {...}` since the if branch is empty but not a big deal.

Reply:
nice, that is much cleaner thanks!